backend/mobilityAI/Controllers/MobilityAIController.cs

```diff
@@ -308,19 +400,99 @@ public IActionResult MlCallback(string Id, IFormFile Activities)
         }
 
         /// <summary>
-        /// Gets processed activity data for a specific device in a specific time range
+        /// Gets processed activity data for a specific device in a specific time range to be displayed on the UI
         /// </summary>
         /// <param name="Start">Epoch for beginning of time range</param>
         /// <param name="End"Epoch for end of time rante></param>
         /// <param name="patientId">Patient Id for the data you want to query</param>
         /// <returns></returns>
         [HttpGet("GetActivityData", Name = "GetActivityData")]
-        public IActionResult GetActivityData(long Start, long End, int patientId)
+        public IActionResult GetActivityData(long start, long end, int patientId)
         {
             var data = (from activities in _context.Activities
-                        where activities.Start >= Start && activities.End <= End && activities.PatientId == patientId
-                        select new { activities.Start, activities.End, activities.Type }).ToList();
-            return Ok(JsonConvert.SerializeObject(data));
+                        where activities.Start >= start && activities.End <= end && activities.PatientId == patientId
+                        orderby activities.Start ascending
+                        select new { activities.Start, activities.End, activities.Type })
+                        .ToList();
+
+            if(data.Count != 0) {
+
+                float[] count = new float[5];
+                float[] total = new float[5];
+                float[][] activityTotals = new float[5][];
+                for (int i = 0; i < 5; i++) activityTotals[i] = new float[13];
+                float totalRows = data.Count;                
+
+                // long firstHour = data[0].Start;
+                // long lastHour = data[data.Count-1].End;
+        
+                // long diff = lastHour - firstHour;
+
+                for(int i = 0; i < data.Count-1; i++) {
+                    var element = data[i];
+                    var nextElement = data[i+1];
+
+                    count[element.Type]++;
+
+                    int startHour = (FromUnixTime(element.Start/1000).Hour) - 12;
+                    activityTotals[element.Type][startHour] += (nextElement.Start - element.Start)/1000f;
+                }
+
+                int startHour2 = (FromUnixTime(data[data.Count-1].Start).Hour) - 12;
+                activityTotals[data[data.Count-1].Type][startHour2] += (data[data.Count-1].End - data[data.Count-1].Start)/1000f;
+
+                for (int i=0; i<5; i++){
+                    for (int j=0; j<13; j++){
+                        activityTotals[i][j] = activityTotals[i][j]/60;
+                    }
+                }
+
+                total[0] = (count[0] / totalRows) * 100;
```

Comment:
Define an `enum Activity { Sitting, ... }` so we don't have magic numbers. eg: `count[Activity.Sitting]`